packages/itwin/tree-widget/src/types.ts

```diff
@@ -5,34 +5,46 @@
 
 import type { SpatialContainmentTreeProps, StagePanelLocation, StagePanelSection } from "@itwin/appui-react";
 import type { SelectableContentDefinition } from "@itwin/components-react";
-import type { IModelConnection } from "@itwin/core-frontend";
+import type { IModelConnection, ScreenViewport } from "@itwin/core-frontend";
+import { IPresentationTreeDataProvider } from "@itwin/presentation-components";
 import type { CategoryTreeProps } from "./components/trees/category-tree/CategoriesTree";
 import type { ModelsTreeProps } from "./components/trees/models-tree/ModelsTree";
 
+export interface TreeHeaderButtonProps {
+  viewport?: ScreenViewport;
+  iModel?: IModelConnection;
```

Comment:
also, could we include the tree model here? we may need that to get selected nodes, etc.

Reply:
In order to include `model` in button props a lot of refactoring is required to move all the state from `ModelsTree` to `ModelsTreeComponent` where header and `ModelsTree` is renderer. I think this could be done in a separate PR.

Reply:
fixed